### DISCUSSION TITLE
QSelect clearable default value?

### DISCUSSION THREAD

**Discussion by pierrecdn** [zauberzeug/nicegui#2475] [Q&A] [answered]
### Question

Hi there,

At reading here and there regarding Quasar's QSelect, I was wondering if I can provide a default value when the select is cleared?

Some highlighted the `clear-value` or `:clear-value` prop in the Quasar community
https://github.com/quasarframework/quasar/issues/2505

But I'm not able to find anything like that and I don't succeed getting anything else than empty working there. 

Here is the user story I'd expect: 
* user defined its own value in the drop down list
* he'd like to reset to defaults; for that he click the "clear" button. Ideally in this case I'd like the "clear" button to disappear, and reappear on selection but that sounds even more complex.

![image](https://github.com/zauberzeug/nicegui/assets/4225738/62bc10a3-7107-459b-a027-2214fb403040)

Any hints?

**Comment by falkoschindler** [comment, marked as answer]
Hi @pierrecdn,

The `:clear-value` prop seems to be outdated. At least the issue https://github.com/quasarframework/quasar/issues/2505 is almost five years old an I can't find it anywhere else either.

Maybe you can build some custom solution like this:
```py
select = ui.select(['default', 'Pasta', 'Pizza'], label='Select', value='default').classes('w-32')
with select.add_slot('append'):
    ui.button(icon='clear', color='grey').props('flat fab-mini').on('click.stop', lambda: select.set_value('default')) \
        .bind_visibility_from(select, 'value', lambda value: value != 'default')
```
